Currency.ts

```diff
@@ -371,9 +371,12 @@ export namespace Currency {
 			"932": "ZWL",
 		}[currencyCode] as Currency
 	}
+
 	export function round(value: number, currency: Currency): number {
 		const factor = Math.pow(10, decimalDigits(currency) ?? 2)
-		return Math.round(Math.round(value * factor * 10) / 10) / factor
+		const decimals = (value.toString().split(".")?.[1]?.length ?? 0) - 1
+		const test = (value + (decimals <= (decimalDigits(currency) ?? 2) + 5 ? 0 : Math.pow(10, -decimals))) * factor
```

Comment:
Please rename "test" into something more describing. Maybe it doesn't need to be a variable?